### PR TITLE
feat(ui): add code editor component

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

- Add code editor component

Capture d’écran 2023-11-24 à 11 14 59

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery  preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery  preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> There are two hard problems in computer science: naming things, cache invalidation and off-by-1 errors

File: libs/shared/ui/src/lib/components/editor/editor.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { Editor as EditMonaco, type EditorProps as EditorMonacoProps } from '@monaco-editor/react'
```
Comment (ctjhoa) [inline libs/shared/ui/src/lib/components/editor/editor.tsx:1]:
```suggestion
import { Editor as EditorMonaco, type EditorProps as EditorMonacoProps } from '@monaco-editor/react'
```

File: libs/shared/ui/src/index.ts [hunk 2]
```diff
@@ -88,6 +88,7 @@ export * from './lib/components/service-icon/service-icon'
 export * from './lib/components/badge-deployment-order/badge-deployment-order'
 export * from './lib/components/badge/badge'
 export * from './lib/components/date-picker/date-picker'
+export * from './lib/components/editor/editor'
```
Comment (ctjhoa) [inline libs/shared/ui/src/index.ts:91]: Can we name it `code-editor` in case we have a text editor in the future?

**Review by ctjhoa — CHANGES_REQUESTED** [review]

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**